EventTarget should hold EventTargetData in WeakPtrImpl

Constellation opened the pull request:
#### d0b2d6ef6a62c77b5b799d332dc2511573931ab8
<pre>
EventTarget should hold EventTargetData in WeakPtrImpl
<a href="https://bugs.webkit.org/show_bug.cgi?id=244710">https://bugs.webkit.org/show_bug.cgi?id=244710</a>

Reviewed by Ryosuke Niwa.

This patch integrates EventTargetData into WeakPtrImpl so that we remove HashMap based side data in Node.
Now, we make WeakPtr&apos;s WeakPtrImpl customizable. For EventTarget, we use WeakPtrImplWithEventTargetData, which embed EventTargetData.
The benefit of this design is that (1) we do not increase sizeof(Node), (2) unifying the location of EventTargetData for all
EventTarget derived classes so we remove virtual function calls for eventTargetData(), and (3) removing HashMap lookup for Node&apos;s EventTargetData,
it is not only slow but also memory consuming and involving costly locking. This new design deploys WTF::storeStoreFence
at the right timing so that we can drop this lock completely.

We also use CompactRefPtrTuple in WeakPtrFactory so we migrate some of flags from Node to WeakPtrFactory&apos;s bitfields.
This includes HasEventTargetData which can tell us that info quickly. We should have this flag separately from WeakPtrFactory&apos;s
WeakPtrImpl&apos;s pointer&apos;s null check because we have many fast path which skips event related thing if HasEventTargetData is false.
We would like to keep this fast path working even if we create WeakPtr for this EventTarget class.

A/B test ensured that this offers 0.37% Speedometer2.1 progression with 98% confidence on AppleSilicon.

* Source/WTF/wtf/WeakHashMap.h:
* Source/WTF/wtf/WeakHashSet.h:
(WTF::copyToVector):
* Source/WTF/wtf/WeakPtr.h:
(WTF::EmptyCounter::increment):
(WTF::EmptyCounter::decrement):
(WTF::WeakPtrImpl::~WeakPtrImpl):
(WTF::WeakPtrImpl::counter):
(WTF::WeakPtrImpl::counter const):
(WTF::WeakPtrImpl::WeakPtrImpl):
(WTF::WeakPtr::implForObject):
(WTF::WeakPtrFactory::~WeakPtrFactory):
(WTF::WeakPtrFactory::counter):
(WTF::WeakPtrFactory::counter const):
(WTF::WeakPtrFactory::initializeIfNeeded const):
(WTF::WeakPtrFactory::createWeakPtr const):
(WTF::WeakPtrFactory::revokeAll):
(WTF::WeakPtrFactory::weakPtrCount const):
(WTF::WeakPtrFactory::isInitialized const):
(WTF::WeakPtrFactory::bitfield const):
(WTF::WeakPtrFactory::setBitfield const):
* Source/WebCore/Modules/WebGPU/GPUDevice.h:
* Source/WebCore/Modules/applepay/ApplePaySession.h:
* Source/WebCore/Modules/applepay/PaymentCoordinator.cpp:
(WebCore::PaymentCoordinator::canMakePaymentsWithActiveCard):
* Source/WebCore/Modules/async-clipboard/Clipboard.h:
* Source/WebCore/Modules/async-clipboard/ClipboardItem.h:
* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.h:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.cpp:
(WebCore::CredentialsContainer::CredentialsContainer):
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.h:
(WebCore::CredentialsContainer::create):
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.cpp:
(WebCore::NavigatorCredentials::credentials):
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
(WebCore::MediaKeySession::generateRequest):
(WebCore::MediaKeySession::load):
(WebCore::MediaKeySession::update):
(WebCore::MediaKeySession::close):
(WebCore::MediaKeySession::remove):
(WebCore::MediaKeySession::stop):
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemAccess.cpp:
(WebCore::MediaKeySystemAccess::createMediaKeys):
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.h:
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeys.h:
* Source/WebCore/Modules/filesystemaccess/WorkerFileSystemStorageConnection.h:
* Source/WebCore/Modules/gamepad/GamepadManager.cpp:
(WebCore::GamepadManager::platformGamepadDisconnected):
(WebCore::GamepadManager::makeGamepadVisible):
* Source/WebCore/Modules/highlight/AppHighlightStorage.h:
* Source/WebCore/Modules/indexeddb/IDBCursor.h:
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
(WebCore::IDBDatabase::dispatchEvent):
* Source/WebCore/Modules/indexeddb/IDBDatabase.h:
* Source/WebCore/Modules/indexeddb/IDBRequest.h:
* Source/WebCore/Modules/indexeddb/IDBTransaction.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/Modules/mediarecorder/MediaRecorder.h:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.h:
* Source/WebCore/Modules/mediasource/MediaSource.h:
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/Modules/mediasource/SourceBufferList.h:
* Source/WebCore/Modules/mediastream/MediaDevices.cpp:
(WebCore::MediaDevices::addEventListener):
* Source/WebCore/Modules/mediastream/MediaDevices.h:
* Source/WebCore/Modules/mediastream/MediaStream.h:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.h:
* Source/WebCore/Modules/mediastream/RTCIceTransport.h:
* Source/WebCore/Modules/mediastream/RTCPeerConnection.h:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
(WebCore::processFrame):
(WebCore::transformFrame):
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpSender.h:
* Source/WebCore/Modules/mediastream/RTCRtpTransceiver.h:
* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
* Source/WebCore/Modules/notifications/Notification.h:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.h:
* Source/WebCore/Modules/paymentrequest/PaymentResponse.h:
* Source/WebCore/Modules/permissions/PermissionStatus.h:
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.h:
* Source/WebCore/Modules/plugins/YouTubePluginReplacement.h:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.h:
* Source/WebCore/Modules/speech/SpeechRecognition.h:
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
* Source/WebCore/Modules/storage/WorkerStorageConnection.h:
* Source/WebCore/Modules/webaudio/AudioNode.cpp:
(WebCore::AudioNode::context):
(WebCore::AudioNode::context const):
* Source/WebCore/Modules/webaudio/AudioNode.h:
* Source/WebCore/Modules/webaudio/AudioSummingJunction.h:
* Source/WebCore/Modules/webaudio/AudioWorklet.h:
* Source/WebCore/Modules/webaudio/BaseAudioContext.h:
* Source/WebCore/Modules/websockets/WebSocket.h:
* Source/WebCore/Modules/websockets/WebSocketChannel.h:
* Source/WebCore/Modules/websockets/WebSocketChannelInspector.h:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXRRenderState.h:
* Source/WebCore/Modules/webxr/WebXRSession.h:
* Source/WebCore/Modules/webxr/WebXRSpace.h:
* Source/WebCore/Modules/webxr/WebXRSystem.h:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::filterWeakHashSetForRemoval):
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AccessibilityListBoxOption.h:
* Source/WebCore/accessibility/AccessibilityMenuListOption.h:
* Source/WebCore/accessibility/AccessibilityScrollView.h:
* Source/WebCore/animation/AnimationEffect.h:
* Source/WebCore/animation/AnimationTimeline.h:
* Source/WebCore/animation/DeclarativeAnimation.h:
* Source/WebCore/animation/DocumentTimeline.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
* Source/WebCore/bindings/js/JSLazyEventListener.h:
* Source/WebCore/css/CSSCursorImageValue.h:
* Source/WebCore/css/CSSFontFaceSource.h:
* Source/WebCore/css/CSSFontFaceSrcValue.h:
* Source/WebCore/css/CSSStyleSheet.h:
* Source/WebCore/css/FontFaceSet.h:
* Source/WebCore/css/MediaQueryEvaluator.h:
* Source/WebCore/css/MediaQueryList.h:
* Source/WebCore/css/MediaQueryMatcher.h:
* Source/WebCore/css/StyleSheetList.h:
* Source/WebCore/css/typedom/CSSStyleImageValue.h:
* Source/WebCore/dom/AbortSignal.cpp:
(WebCore::AbortSignal::signalFollow):
* Source/WebCore/dom/AbortSignal.h:
* Source/WebCore/dom/Attr.h:
* Source/WebCore/dom/BroadcastChannel.cpp:
* Source/WebCore/dom/BroadcastChannel.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::buildAccessKeyCache):
(WebCore::Document::determineActiveThemeColorMetaElement):
(WebCore::Document::didLogMessage):
(WebCore::Document::navigateFromServiceWorker):
(WebCore::Document::didRejectSyncXHRDuringPageDismissal):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentParser.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::setElementAttribute):
(WebCore::Element::setElementsArrayAttribute):
(WebCore::elementIdentifiersMap):
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementInternals.h:
* Source/WebCore/dom/EventListenerMap.h:
(WebCore::EventListenerMap::clearEntriesForTearDown):
* Source/WebCore/dom/EventSender.h:
(WebCore::Counter&gt;::EventSender):
(WebCore::Counter&gt;::dispatchEventSoon):
(WebCore::Counter&gt;::cancelEvent):
(WebCore::Counter&gt;::dispatchPendingEvents):
(WebCore::EventSender&lt;T&gt;::EventSender): Deleted.
(WebCore::EventSender&lt;T&gt;::dispatchEventSoon): Deleted.
(WebCore::EventSender&lt;T&gt;::cancelEvent): Deleted.
(WebCore::EventSender&lt;T&gt;::dispatchPendingEvents): Deleted.
* Source/WebCore/dom/EventTarget.cpp:
* Source/WebCore/dom/EventTarget.h:
(WebCore::EventTargetData::clear):
(WebCore::EventTargetCounter::increment):
(WebCore::EventTargetCounter::decrement):
(WebCore::EventTarget::eventTargetData const):
(WebCore::EventTarget::eventTargetData):
(WebCore::EventTarget::eventTargetDataConcurrently):
(WebCore::EventTarget::hasEventTargetData const):
(WebCore::EventTarget::setHasEventTargetData const):
(WebCore::EventTarget::ensureEventTargetData):
(): Deleted.
* Source/WebCore/dom/EventTargetConcrete.h:
* Source/WebCore/dom/IdleCallbackController.h:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::addEventListener):
(WebCore::MessagePort::removeEventListener):
* Source/WebCore/dom/MessagePort.h:
* Source/WebCore/dom/Node.cpp:
(WebCore::liveNodeSet):
(WebCore::ignoreSet):
(WebCore::Node::clearEventTargetData):
(WebCore::eventTargetDataMap): Deleted.
(WebCore::Node::eventTargetData): Deleted.
(WebCore::Node::eventTargetDataConcurrently): Deleted.
(WebCore::Node::ensureEventTargetData): Deleted.
* Source/WebCore/dom/Node.h:
(WebCore::Node::hasEventTargetData const): Deleted.
(WebCore::Node::setHasEventTargetData): Deleted.
* Source/WebCore/dom/NodeRareData.cpp:
* Source/WebCore/dom/NodeRareData.h:
* Source/WebCore/dom/PseudoElement.h:
* Source/WebCore/dom/RadioButtonGroups.cpp:
* Source/WebCore/dom/ScriptedAnimationController.h:
* Source/WebCore/dom/ShadowRoot.cpp:
(WebCore::ShadowRoot::slotManualAssignmentDidChange):
(WebCore::ShadowRoot::assignedNodesForSlot):
* Source/WebCore/dom/ShadowRoot.h:
* Source/WebCore/dom/SlotAssignment.cpp:
(WebCore::NamedSlotAssignment::slotManualAssignmentDidChange):
(WebCore::NamedSlotAssignment::assignedNodesForSlot):
(WebCore::effectiveAssignedNodes):
(WebCore::ManualSlotAssignment::assignedNodesForSlot):
(WebCore::ManualSlotAssignment::slotManualAssignmentDidChange):
* Source/WebCore/dom/SlotAssignment.h:
* Source/WebCore/dom/TemplateContentDocumentFragment.h:
* Source/WebCore/dom/UserGestureIndicator.h:
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::containsEndpoints):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/TextManipulationController.h:
* Source/WebCore/fileapi/FileReader.h:
* Source/WebCore/html/FormAssociatedElement.h:
* Source/WebCore/html/FormController.cpp:
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::rootEditableElementMap):
* Source/WebCore/html/HTMLDetailsElement.h:
* Source/WebCore/html/HTMLDialogElement.h:
* Source/WebCore/html/HTMLFieldSetElement.h:
* Source/WebCore/html/HTMLFormControlsCollection.cpp:
(WebCore::findFormAssociatedElement):
* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::unsafeAssociatedElements const):
* Source/WebCore/html/HTMLFormElement.h:
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLLinkElement.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::updateActiveTextTrackCues):
(WebCore::HTMLMediaElement::speakCueText):
(WebCore::HTMLMediaElement::progressEventTimerFired):
* Source/WebCore/html/HTMLSlotElement.cpp:
(WebCore::HTMLSlotElement::assignedNodes const):
(WebCore::HTMLSlotElement::assign):
* Source/WebCore/html/HTMLSlotElement.h:
* Source/WebCore/html/HTMLStyleElement.h:
* Source/WebCore/html/ImageDocument.h:
* Source/WebCore/html/InputType.h:
* Source/WebCore/html/MediaController.h:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/ModelDocument.cpp:
* Source/WebCore/html/OffscreenCanvas.cpp:
(WebCore::DetachedOffscreenCanvas::takePlaceholderCanvas):
* Source/WebCore/html/OffscreenCanvas.h:
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/ValidationMessage.h:
* Source/WebCore/html/parser/HTMLScriptRunner.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.h:
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TextTrackCue.h:
* Source/WebCore/html/track/TrackBase.h:
* Source/WebCore/html/track/TrackListBase.h:
* Source/WebCore/inspector/InspectorOverlay.cpp:
(WebCore::InspectorOverlay::removeGridOverlayForNode):
(WebCore::InspectorOverlay::removeFlexOverlayForNode):
* Source/WebCore/inspector/InspectorOverlay.h:
* Source/WebCore/inspector/agents/InspectorCSSAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMAgent.h:
* Source/WebCore/inspector/agents/InspectorLayerTreeAgent.h:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::enable):
* Source/WebCore/loader/ImageLoader.h:
* Source/WebCore/loader/LinkLoader.cpp:
(WebCore::LinkLoader::preconnectIfNeeded):
* Source/WebCore/loader/MediaResourceLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheHost.h:
* Source/WebCore/loader/appcache/DOMApplicationCache.h:
* Source/WebCore/loader/cache/CachedImage.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/page/AbstractDOMWindow.h:
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::windowsInterestedInStorageEvents):
* Source/WebCore/page/DOMWindowExtension.h:
* Source/WebCore/page/DOMWindowProperty.h:
* Source/WebCore/page/EventSource.h:
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/ImageAnalysisQueue.h:
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/IntersectionObserver.h:
(WebCore::IntersectionObserver::observationTargets const):
* Source/WebCore/page/ModalContainerObserver.cpp:
(WebCore::ModalContainerObserver::collectClickableElementsTimerFired):
(WebCore::ModalContainerObserver::collectClickableElements):
* Source/WebCore/page/ModalContainerObserver.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateRendering):
(WebCore::Page::doAfterUpdateRendering):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/Performance.h:
* Source/WebCore/page/PointerLockController.h:
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/ResizeObservation.h:
* Source/WebCore/page/ResizeObserver.h:
* Source/WebCore/page/UndoItem.h:
* Source/WebCore/page/VisualViewport.h:
* Source/WebCore/page/ios/ContentChangeObserver.h:
* Source/WebCore/page/scrolling/ScrollAnchoringController.h:
* Source/WebCore/page/scrolling/ScrollLatchingController.h:
* Source/WebCore/rendering/svg/RenderSVGResource.cpp:
(WebCore::removeFromCacheAndInvalidateDependencies):
* Source/WebCore/style/StyleScope.h:
* Source/WebCore/svg/SVGDocumentExtensions.cpp:
(WebCore::SVGDocumentExtensions::addPendingResource):
* Source/WebCore/svg/SVGDocumentExtensions.h:
(WebCore::SVGDocumentExtensions::svgFontFaceElements const):
(WebCore::SVGDocumentExtensions::removePendingResource):
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::instances const):
* Source/WebCore/svg/SVGElement.h:
* Source/WebCore/svg/SVGElementRareData.h:
(WebCore::SVGElementRareData::instances const):
(WebCore::SVGElementRareData::referencingElements const):
(WebCore::SVGElementRareData::takeReferencingElements):
(WebCore::SVGElementRareData::setReferenceTarget):
* Source/WebCore/svg/SVGFontFaceElement.h:
* Source/WebCore/svg/SVGViewElement.h:
* Source/WebCore/svg/SVGViewSpec.h:
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::notifyDependentsIntervalChanged):
* Source/WebCore/svg/animation/SVGSMILElement.h:
* Source/WebCore/testing/WebXRTest.h:
* Source/WebCore/workers/AbstractWorker.h:
* Source/WebCore/workers/WorkerOrWorkletGlobalScope.h:
* Source/WebCore/workers/WorkerRunLoop.h:
* Source/WebCore/workers/service/ServiceWorker.h:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
(WebCore::ServiceWorkerContainer::addEventListener):
* Source/WebCore/workers/service/ServiceWorkerContainer.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.h:
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/WebCore/xml/DOMParser.h:
* Source/WebCore/xml/XMLHttpRequestEventTarget.h:
* Source/WebCore/xml/XSLStyleSheet.h:
* Source/WebKit/WebProcess/Automation/WebAutomationDOMWindowObserver.h:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.h:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/DOMObjectCache.cpp:
* Source/WebKit/WebProcess/Network/WebSocketChannel.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/cocoa/VideoFullscreenManager.h:

Canonical link: <a href="https://commits.webkit.org/254173@main">https://commits.webkit.org/254173@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7a7f1af6bc47544ff716bbd4739f3f74e0a47d1b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88307 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/32693 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19120 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/97502 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/152968 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92274 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31190 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26869 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/80503 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92158 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93916 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/24846 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75117 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/24823 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79753 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79872 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67789 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80043 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/28778 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/13844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/73802 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/28778 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14855 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26184 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2930 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/31940 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37766 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/76654 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/30935 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33990 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17009 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WTF/wtf/WeakHashSet.h:
```diff
@@ -116,9 +118,11 @@ class WeakHashSet final {
-        if (!weakPtrImpl || !*weakPtrImpl)
-            return false;
-        return m_set.contains(*weakPtrImpl);
+        if (auto* pointer = weakPtrImpl.pointer()) {
```
Ditto.

Constellation replied:
Changed.

rniwa commented on Source/WTF/wtf/WeakHashMap.h:
```diff
@@ -369,9 +369,11 @@ class WeakHashMap final {
-        if (!weakPtrImpl || !*weakPtrImpl)
-            return nullptr;
-        return weakPtrImpl.get();
+        if (auto* pointer = weakPtrImpl.pointer()) {
```
Why not: auto* pointer = weakPtrImpl.pointer(); pointer && *pointer?

Constellation replied:
That sounds good. Changed.

rniwa commented on Source/WTF/wtf/WeakHashSet.h:
```diff
@@ -105,9 +105,11 @@ class WeakHashSet final {
-            return false;
-        return m_set.remove(*weakPtrImpl);
+        if (auto* pointer = weakPtrImpl.pointer()) {
+            if (*pointer)
```
Ditto.

Constellation replied:
Changed.

rniwa commented on Source/WTF/wtf/WeakPtr.h:
```diff
@@ -34,8 +35,8 @@
-    static void decrement() { }
+    void increment() { }
+    void decrement() { }
 };
```
Since this counter thing is only useful in testing, we can put this in the specialized version of WeakPtrImpl
the same way we put EventTargetData in WeakPtrImpl.

Constellation replied:
Done.

rniwa commented on Source/WTF/wtf/WeakPtr.h:
```diff
@@ -200,49 +205,69 @@ template<typename T, typename Counter = EmptyCounter> class WeakPtrFactory {
     template<typename, typename> friend class WeakPtr;
 
-    mutable RefPtr<WeakPtrImpl<Counter>> m_impl;
+    mutable CompactRefPtrTuple<WeakPtrImpl<CounterType>, uint16_t> m_impl;
```
The way I was thinking of this, we can specify WeakPtrImpl type the same way we specify T.
So, WeakPtr usage becomes WeakPtr<T, WeakPtrImpl = DefaultWeakPtrImpl>.
Then WeakPtrImpl can be specialized for EventTarget.

Constellation replied:
Sounds good.

rniwa commented on Source/WebCore/Modules/WebGPU/GPUDevice.h:
```diff
@@ -75,7 +75,7 @@ class GPUSupportedLimits;
 class GPUTexture;
 struct GPUTextureDescriptor;
 
-class GPUDevice : public RefCounted<GPUDevice>, public ActiveDOMObject, public EventTargetWithInlineData {
```
For simplicity, why don't we alias EventTargetWithInlineData with EventTarget for now.
Once this patch landed, we can clean that up as a follow up.

Constellation replied:
Since this is mechanically changeable, this patch also just changed all EventTargetWithInlineData with EventTarget.

rniwa commented on Source/WebCore/dom/EventTarget.h:
```diff
@@ -50,16 +50,29 @@ namespace WebCore {
 };
 
-class EventTarget : public ScriptWrappable, public CanMakeWeakPtr<EventTarget> {
+class EventTargetCounter : public EventTargetData {
```
I'd call this EventTargetDataWithCounter.
But really, I don't think we even need to have this counter concept in most WeakPtrImpl.
We can have this defined only in a specialized version for the tests.

Constellation replied:
Renamed to `WeakPtrImplWithEventTargetData`.

rniwa commented on Source/WTF/wtf/WeakHashSet.h:
```diff
@@ -116,9 +118,11 @@ class WeakHashSet final {
-            return false;
-        return m_set.contains(*weakPtrImpl);
+        if (auto* pointer = weakPtrImpl.pointer()) {
+            if (*pointer)
```
Ditto.

Constellation replied:
Changed.

rniwa commented on Source/WTF/wtf/WeakPtr.h:
```diff
@@ -76,10 +80,11 @@ template<typename Counter = EmptyCounter> class WeakPtrImpl : public ThreadSafeR
     }
 
     void* m_ptr;
+    NO_UNIQUE_ADDRESS Counter m_counter;
```
Rather than including Counter as a member, we can specialize WeakPtrImpl and store data there.

Constellation replied:
OK, apply this change.

rniwa reviewed (approved):
r=me. This is okay but I still think an alternative design of specializing WeakPtrImpl and only having counter in the test is a cleaner design.

Constellation committed:
EventTarget should hold EventTargetData in WeakPtrImpl
https://bugs.webkit.org/show_bug.cgi?id=244710

Reviewed by Ryosuke Niwa.

This patch integrates EventTargetData into WeakPtrImpl so that we remove HashMap based side data in Node.
Now, we make WeakPtr's WeakPtrImpl customizable. For EventTarget, we use WeakPtrImplWithEventTargetData, which embed EventTargetData.
The benefit of this design is that (1) we do not increase sizeof(Node), (2) unifying the location of EventTargetData for all
EventTarget derived classes so we remove virtual function calls for eventTargetData(), and (3) removing HashMap lookup for Node's EventTargetData,
it is not only slow but also memory consuming and involving costly locking. This new design deploys WTF::storeStoreFence
at the right timing so that we can drop this lock completely.

We also use CompactRefPtrTuple in WeakPtrFactory so we migrate some of flags from Node to WeakPtrFactory's bitfields.
This includes HasEventTargetData which can tell us that info quickly. We should have this flag separately from WeakPtrFactory's
WeakPtrImpl's pointer's null check because we have many fast path which skips event related thing if HasEventTargetData is false.
We would like to keep this fast path working even if we create WeakPtr for this EventTarget class.

A/B test ensured that this offers 0.37% Speedometer2.1 progression with 98% confidence on AppleSilicon.

* Source/WTF/wtf/WeakHashMap.h:
* Source/WTF/wtf/WeakHashSet.h:
(WTF::copyToVector):
* Source/WTF/wtf/WeakPtr.h:
(WTF::EmptyCounter::increment):
(WTF::EmptyCounter::decrement):
(WTF::WeakPtrImpl::~WeakPtrImpl):
(WTF::WeakPtrImpl::counter):
(WTF::WeakPtrImpl::counter const):
(WTF::WeakPtrImpl::WeakPtrImpl):
(WTF::WeakPtr::implForObject):
(WTF::WeakPtrFactory::~WeakPtrFactory):
(WTF::WeakPtrFactory::counter):
(WTF::WeakPtrFactory::counter const):
(WTF::WeakPtrFactory::initializeIfNeeded const):
(WTF::WeakPtrFactory::createWeakPtr const):
(WTF::WeakPtrFactory::revokeAll):
(WTF::WeakPtrFactory::weakPtrCount const):
(WTF::WeakPtrFactory::isInitialized const):
(WTF::WeakPtrFactory::bitfield const):
(WTF::WeakPtrFactory::setBitfield const):
* Source/WebCore/Modules/WebGPU/GPUDevice.h:
* Source/WebCore/Modules/applepay/ApplePaySession.h:
* Source/WebCore/Modules/applepay/PaymentCoordinator.cpp:
(WebCore::PaymentCoordinator::canMakePaymentsWithActiveCard):
* Source/WebCore/Modules/async-clipboard/Clipboard.h:
* Source/WebCore/Modules/async-clipboard/ClipboardItem.h:
* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.h:
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.cpp:
(WebCore::CredentialsContainer::CredentialsContainer):
* Source/WebCore/Modules/credentialmanagement/CredentialsContainer.h:
(WebCore::CredentialsContainer::create):
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.cpp:
(WebCore::NavigatorCredentials::credentials):
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
(WebCore::MediaKeySession::generateRequest):
(WebCore::MediaKeySession::load):
(WebCore::MediaKeySession::update):
(WebCore::MediaKeySession::close):
(WebCore::MediaKeySession::remove):
(WebCore::MediaKeySession::stop):
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemAccess.cpp:
(WebCore::MediaKeySystemAccess::createMediaKeys):
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.h:
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeys.h:
* Source/WebCore/Modules/filesystemaccess/WorkerFileSystemStorageConnection.h:
* Source/WebCore/Modules/gamepad/GamepadManager.cpp:
(WebCore::GamepadManager::platformGamepadDisconnected):
(WebCore::GamepadManager::makeGamepadVisible):
* Source/WebCore/Modules/highlight/AppHighlightStorage.h:
* Source/WebCore/Modules/indexeddb/IDBCursor.h:
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
(WebCore::IDBDatabase::dispatchEvent):
* Source/WebCore/Modules/indexeddb/IDBDatabase.h:
* Source/WebCore/Modules/indexeddb/IDBRequest.h:
* Source/WebCore/Modules/indexeddb/IDBTransaction.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/Modules/mediarecorder/MediaRecorder.h:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.h:
* Source/WebCore/Modules/mediasource/MediaSource.h:
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/Modules/mediasource/SourceBufferList.h:
* Source/WebCore/Modules/mediastream/MediaDevices.cpp:
(WebCore::MediaDevices::addEventListener):
* Source/WebCore/Modules/mediastream/MediaDevices.h:
* Source/WebCore/Modules/mediastream/MediaStream.h:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.h:
* Source/WebCore/Modules/mediastream/RTCDataChannel.h:
* Source/WebCore/Modules/mediastream/RTCDtlsTransport.h:
* Source/WebCore/Modules/mediastream/RTCIceTransport.h:
* Source/WebCore/Modules/mediastream/RTCPeerConnection.h:
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.cpp:
(WebCore::processFrame):
(WebCore::transformFrame):
* Source/WebCore/Modules/mediastream/RTCRtpSFrameTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpSender.h:
* Source/WebCore/Modules/mediastream/RTCRtpTransceiver.h:
* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
* Source/WebCore/Modules/notifications/Notification.h:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.h:
* Source/WebCore/Modules/paymentrequest/PaymentResponse.h:
* Source/WebCore/Modules/permissions/PermissionStatus.h:
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.h:
* Source/WebCore/Modules/plugins/YouTubePluginReplacement.h:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.h:
* Source/WebCore/Modules/speech/SpeechRecognition.h:
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
* Source/WebCore/Modules/storage/WorkerStorageConnection.h:
* Source/WebCore/Modules/webaudio/AudioNode.cpp:
(WebCore::AudioNode::context):
(WebCore::AudioNode::context const):
* Source/WebCore/Modules/webaudio/AudioNode.h:
* Source/WebCore/Modules/webaudio/AudioSummingJunction.h:
* Source/WebCore/Modules/webaudio/AudioWorklet.h:
* Source/WebCore/Modules/webaudio/BaseAudioContext.h:
* Source/WebCore/Modules/websockets/WebSocket.h:
* Source/WebCore/Modules/websockets/WebSocketChannel.h:
* Source/WebCore/Modules/websockets/WebSocketChannelInspector.h:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXRRenderState.h:
* Source/WebCore/Modules/webxr/WebXRSession.h:
* Source/WebCore/Modules/webxr/WebXRSpace.h:
* Source/WebCore/Modules/webxr/WebXRSystem.h:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::filterWeakHashSetForRemoval):
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AccessibilityListBoxOption.h:
* Source/WebCore/accessibility/AccessibilityMenuListOption.h:
* Source/WebCore/accessibility/AccessibilityScrollView.h:
* Source/WebCore/animation/AnimationEffect.h:
* Source/WebCore/animation/AnimationTimeline.h:
* Source/WebCore/animation/DeclarativeAnimation.h:
* Source/WebCore/animation/DocumentTimeline.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
* Source/WebCore/bindings/js/JSLazyEventListener.h:
* Source/WebCore/css/CSSCursorImageValue.h:
* Source/WebCore/css/CSSFontFaceSource.h:
* Source/WebCore/css/CSSFontFaceSrcValue.h:
* Source/WebCore/css/CSSStyleSheet.h:
* Source/WebCore/css/FontFaceSet.h:
* Source/WebCore/css/MediaQueryEvaluator.h:
* Source/WebCore/css/MediaQueryList.h:
* Source/WebCore/css/MediaQueryMatcher.h:
* Source/WebCore/css/StyleSheetList.h:
* Source/WebCore/css/typedom/CSSStyleImageValue.h:
* Source/WebCore/dom/AbortSignal.cpp:
(WebCore::AbortSignal::signalFollow):
* Source/WebCore/dom/AbortSignal.h:
* Source/WebCore/dom/Attr.h:
* Source/WebCore/dom/BroadcastChannel.cpp:
* Source/WebCore/dom/BroadcastChannel.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::buildAccessKeyCache):
(WebCore::Document::determineActiveThemeColorMetaElement):
(WebCore::Document::didLogMessage):
(WebCore::Document::navigateFromServiceWorker):
(WebCore::Document::didRejectSyncXHRDuringPageDismissal):
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentParser.h:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::setElementAttribute):
(WebCore::Element::setElementsArrayAttribute):
(WebCore::elementIdentifiersMap):
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementInternals.h:
* Source/WebCore/dom/EventListenerMap.h:
(WebCore::EventListenerMap::clearEntriesForTearDown):
* Source/WebCore/dom/EventSender.h:
(WebCore::Counter>::EventSender):
(WebCore::Counter>::dispatchEventSoon):
(WebCore::Counter>::cancelEvent):
(WebCore::Counter>::dispatchPendingEvents):
(WebCore::EventSender<T>::EventSender): Deleted.
(WebCore::EventSender<T>::dispatchEventSoon): Deleted.
(WebCore::EventSender<T>::cancelEvent): Deleted.
(WebCore::EventSender<T>::dispatchPendingEvents): Deleted.
* Source/WebCore/dom/EventTarget.cpp:
* Source/WebCore/dom/EventTarget.h:
(WebCore::EventTargetData::clear):
(WebCore::EventTargetCounter::increment):
(WebCore::EventTargetCounter::decrement):
(WebCore::EventTarget::eventTargetData const):
(WebCore::EventTarget::eventTargetData):
(WebCore::EventTarget::eventTargetDataConcurrently):
(WebCore::EventTarget::hasEventTargetData const):
(WebCore::EventTarget::setHasEventTargetData const):
(WebCore::EventTarget::ensureEventTargetData):
(): Deleted.
* Source/WebCore/dom/EventTargetConcrete.h:
* Source/WebCore/dom/IdleCallbackController.h:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::addEventListener):
(WebCore::MessagePort::removeEventListener):
* Source/WebCore/dom/MessagePort.h:
* Source/WebCore/dom/Node.cpp:
(WebCore::liveNodeSet):
(WebCore::ignoreSet):
(WebCore::Node::clearEventTargetData):
(WebCore::eventTargetDataMap): Deleted.
(WebCore::Node::eventTargetData): Deleted.
(WebCore::Node::eventTargetDataConcurrently): Deleted.
(WebCore::Node::ensureEventTargetData): Deleted.
* Source/WebCore/dom/Node.h:
(WebCore::Node::hasEventTargetData const): Deleted.
(WebCore::Node::setHasEventTargetData): Deleted.
* Source/WebCore/dom/NodeRareData.cpp:
* Source/WebCore/dom/NodeRareData.h:
* Source/WebCore/dom/PseudoElement.h:
* Source/WebCore/dom/RadioButtonGroups.cpp:
* Source/WebCore/dom/ScriptedAnimationController.h:
* Source/WebCore/dom/ShadowRoot.cpp:
(WebCore::ShadowRoot::slotManualAssignmentDidChange):
(WebCore::ShadowRoot::assignedNodesForSlot):
* Source/WebCore/dom/ShadowRoot.h:
* Source/WebCore/dom/SlotAssignment.cpp:
(WebCore::NamedSlotAssignment::slotManualAssignmentDidChange):
(WebCore::NamedSlotAssignment::assignedNodesForSlot):
(WebCore::effectiveAssignedNodes):
(WebCore::ManualSlotAssignment::assignedNodesForSlot):
(WebCore::ManualSlotAssignment::slotManualAssignmentDidChange):
* Source/WebCore/dom/SlotAssignment.h:
* Source/WebCore/dom/TemplateContentDocumentFragment.h:
* Source/WebCore/dom/UserGestureIndicator.h:
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::containsEndpoints):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/editing/TextManipulationController.h:
* Source/WebCore/fileapi/FileReader.h:
* Source/WebCore/html/FormAssociatedElement.h:
* Source/WebCore/html/FormController.cpp:
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::rootEditableElementMap):
* Source/WebCore/html/HTMLDetailsElement.h:
* Source/WebCore/html/HTMLDialogElement.h:
* Source/WebCore/html/HTMLFieldSetElement.h:
* Source/WebCore/html/HTMLFormControlsCollection.cpp:
(WebCore::findFormAssociatedElement):
* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::unsafeAssociatedElements const):
* Source/WebCore/html/HTMLFormElement.h:
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLLinkElement.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::updateActiveTextTrackCues):
(WebCore::HTMLMediaElement::speakCueText):
(WebCore::HTMLMediaElement::progressEventTimerFired):
* Source/WebCore/html/HTMLSlotElement.cpp:
(WebCore::HTMLSlotElement::assignedNodes const):
(WebCore::HTMLSlotElement::assign):
* Source/WebCore/html/HTMLSlotElement.h:
* Source/WebCore/html/HTMLStyleElement.h:
* Source/WebCore/html/ImageDocument.h:
* Source/WebCore/html/InputType.h:
* Source/WebCore/html/MediaController.h:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/ModelDocument.cpp:
* Source/WebCore/html/OffscreenCanvas.cpp:
(WebCore::DetachedOffscreenCanvas::takePlaceholderCanvas):
* Source/WebCore/html/OffscreenCanvas.h:
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/ValidationMessage.h:
* Source/WebCore/html/parser/HTMLScriptRunner.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.h:
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TextTrackCue.h:
* Source/WebCore/html/track/TrackBase.h:
* Source/WebCore/html/track/TrackListBase.h:
* Source/WebCore/inspector/InspectorOverlay.cpp:
(WebCore::InspectorOverlay::removeGridOverlayForNode):
(WebCore::InspectorOverlay::removeFlexOverlayForNode):
* Source/WebCore/inspector/InspectorOverlay.h:
* Source/WebCore/inspector/agents/InspectorCSSAgent.h:
* Source/WebCore/inspector/agents/InspectorDOMAgent.h:
* Source/WebCore/inspector/agents/InspectorLayerTreeAgent.h:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::enable):
* Source/WebCore/loader/ImageLoader.h:
* Source/WebCore/loader/LinkLoader.cpp:
(WebCore::LinkLoader::preconnectIfNeeded):
* Source/WebCore/loader/MediaResourceLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheHost.h:
* Source/WebCore/loader/appcache/DOMApplicationCache.h:
* Source/WebCore/loader/cache/CachedImage.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/page/AbstractDOMWindow.h:
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::windowsInterestedInStorageEvents):
* Source/WebCore/page/DOMWindowExtension.h:
* Source/WebCore/page/DOMWindowProperty.h:
* Source/WebCore/page/EventSource.h:
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/ImageAnalysisQueue.h:
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/IntersectionObserver.h:
(WebCore::IntersectionObserver::observationTargets const):
* Source/WebCore/page/ModalContainerObserver.cpp:
(WebCore::ModalContainerObserver::collectClickableElementsTimerFired):
(WebCore::ModalContainerObserver::collectClickableElements):
* Source/WebCore/page/ModalContainerObserver.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateRendering):
(WebCore::Page::doAfterUpdateRendering):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/Performance.h:
* Source/WebCore/page/PointerLockController.h:
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/ResizeObservation.h:
* Source/WebCore/page/ResizeObserver.h:
* Source/WebCore/page/UndoItem.h:
* Source/WebCore/page/VisualViewport.h:
* Source/WebCore/page/ios/ContentChangeObserver.h:
* Source/WebCore/page/scrolling/ScrollAnchoringController.h:
* Source/WebCore/page/scrolling/ScrollLatchingController.h:
* Source/WebCore/rendering/svg/RenderSVGResource.cpp:
(WebCore::removeFromCacheAndInvalidateDependencies):
* Source/WebCore/style/StyleScope.h:
* Source/WebCore/svg/SVGDocumentExtensions.cpp:
(WebCore::SVGDocumentExtensions::addPendingResource):
* Source/WebCore/svg/SVGDocumentExtensions.h:
(WebCore::SVGDocumentExtensions::svgFontFaceElements const):
(WebCore::SVGDocumentExtensions::removePendingResource):
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::instances const):
* Source/WebCore/svg/SVGElement.h:
* Source/WebCore/svg/SVGElementRareData.h:
(WebCore::SVGElementRareData::instances const):
(WebCore::SVGElementRareData::referencingElements const):
(WebCore::SVGElementRareData::takeReferencingElements):
(WebCore::SVGElementRareData::setReferenceTarget):
* Source/WebCore/svg/SVGFontFaceElement.h:
* Source/WebCore/svg/SVGViewElement.h:
* Source/WebCore/svg/SVGViewSpec.h:
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::notifyDependentsIntervalChanged):
* Source/WebCore/svg/animation/SVGSMILElement.h:
* Source/WebCore/testing/WebXRTest.h:
* Source/WebCore/workers/AbstractWorker.h:
* Source/WebCore/workers/WorkerOrWorkletGlobalScope.h:
* Source/WebCore/workers/WorkerRunLoop.h:
* Source/WebCore/workers/service/ServiceWorker.h:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
(WebCore::ServiceWorkerContainer::addEventListener):
* Source/WebCore/workers/service/ServiceWorkerContainer.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.h:
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/WebCore/xml/DOMParser.h:
* Source/WebCore/xml/XMLHttpRequestEventTarget.h:
* Source/WebCore/xml/XSLStyleSheet.h:
* Source/WebKit/WebProcess/Automation/WebAutomationDOMWindowObserver.h:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.h:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/DOMObjectCache.cpp:
* Source/WebKit/WebProcess/Network/WebSocketChannel.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/cocoa/VideoFullscreenManager.h:

Canonical link: https://commits.webkit.org/254173@main
